subprojects/docs/src/docs/release/notes.md

```diff
@@ -21,151 +21,32 @@ Switch your build to use Gradle @version@ by updating your wrapper properties:
+Up to Gradle 5.4, for implementing an incremental task you would use [`IncrementalTaskInputs`](dsl/org.gradle.api.tasks.incremental.IncrementalTaskInputs.html).
 
-```
-import com.google.common.hash.Hashing
-import com.google.common.io.Files
-import org.gradle.api.DefaultTask
-import org.gradle.api.file.ConfigurableFileCollection
-import org.gradle.api.file.DirectoryProperty
-import org.gradle.api.file.FileType
-import org.gradle.api.tasks.InputFiles
-import org.gradle.api.tasks.OutputDirectory
-import org.gradle.api.tasks.PathSensitive
-import org.gradle.api.tasks.PathSensitivity
-import org.gradle.api.tasks.SkipWhenEmpty
-import org.gradle.api.tasks.TaskAction
-import org.gradle.work.ChangeType
-import org.gradle.work.FileChange
-import org.gradle.work.InputChanges
-
-class Checksum extends DefaultTask {
-    @SkipWhenEmpty
-    @PathSensitive(PathSensitivity.NAME_ONLY)
-    @InputFiles
-    ConfigurableFileCollection sources = project.files()
-
-    @OutputDirectory
-    DirectoryProperty outputDirectory = project.objects.directoryProperty()
-
-    @TaskAction
-    void checksum(InputChanges changes) {
-        File outputDir = outputDirectory.get().asFile
-        if (!changes.incremental) {
-            println("Non-incremental changes - cleaning output directory")
-            outputDir.deleteDir()
-            outputDir.mkdirs()
-        }
-        for(FileChange change : changes.getFileChanges(sources)) {
-            File changedFile = change.file
-            if (change.fileType == FileType.FILE) {
-                switch(change.changeType) {
-                    case ChangeType.ADDED: // fall through
-                    case ChangeType.MODIFIED:
-                        outputFileForInput(change).text = hashFileContents(changedFile)
-                        break
-                    case ChangeType.REMOVED:
-                        deleteStaleOutput(outputFileForInput(change))
-                        break
-                }
-            }
-        }
-    }
-
-    private void deleteStaleOutput( File file) {
-        println("Removing old output ${file.name}")
-        file.delete()
-    }
+Now you can use the new [`InputChanges`](dsl/org.gradle.work.InputChanges.html) API for implementing incremental tasks.
+This API addresses some shortcomings of the old API, first and foremost that it is now possible to query for changes of individual input file properties, instead of receiving the changes for all input file properties at once.
+Additionally, the file type and the normalized path can be queried for each change, and the old outputs of the task are automatically removed on non-incremental execution.
 
-    private File outputFileForInput(FileChange inputFile) {
-        project.file("${outputDirectory.get().asFile}/${inputFile.normalizedPath}.sha256")
-    }
+See the [userguide section](userguide/custom_tasks.html#incremental_tasks) for more information on how to implement incremental tasks using the new API.
```

Comment:
Please use "user manual" rather than "user guide" in visible text. The URLs are unlikely to change at this point, but we do want to make sure that users always read "user manual".

Reply:
Changed in https://github.com/gradle/gradle/pull/8952.